src/render/SDL_render.c

```diff
@@ -2984,7 +2984,130 @@ SDL_RenderDrawLinesF(SDL_Renderer * renderer,
         return RenderDrawLinesWithRectsF(renderer, points, count);
     }
 
-    retval = QueueCmdDrawLines(renderer, points, count);
+    if (!(renderer->info.flags & SDL_RENDERER_SOFTWARE) && renderer->QueueGeometry) {
+        SDL_bool isstack1;
+        SDL_bool isstack2;
+        float *xy = SDL_small_alloc(float, 4 * 2 * count, &isstack1);
+        int *indices = SDL_small_alloc(int,
+                  (4) * 3 * (count - 1)
+                + (2) * 3 * (count)
+                , &isstack2);
+
+        if (xy && indices) {
+            int i;
+            float *ptr_xy = xy;
+            int *ptr_indices = indices;
+            const int xy_stride = 2 * sizeof (float);
+            int num_vertices = 4 * count;
+            int num_indices = 0;
+            const int size_indices = 4;
+            int cur_indice = -4;
+            const int is_looping = (points[0].x == points[count - 1].x && points[0].y == points[count - 1].y);
+            SDL_FPoint p; /* previous point */
+            p.x = p.y = 0.0f;
+            /*       p            q
+
+                    0----1------ 4----5
+                    | \  |``\    | \  |
+                    |  \ |   ` `\|  \ |
+                    3----2-------7----6
+            */
+            for (i = 0; i < count; ++i) {
+                SDL_FPoint q; /* current point */
+
+                q = points[i];
+                q.x += 0.5f;
+                q.y += 0.5f;
+
+                *ptr_xy++ = SDL_floorf(q.x);
```

Comment:
Since this is an `F`-suffixed API (accepts floats and renders at subpixel precision according to the header docs), are these `floor` calls useful? I don't know how the old implementation behaved in that situation.

I did some profiling on this function in Windows using Visual Studio's CPU profiler a little while ago and `SDL_floorf` is pretty expensive when SDL's own implementation is used instead of the CRT implementation.

If the floor calls aren't needed then removing them all will be a fairly substantial performance improvement (on Windows at least). And if they are necessary, then this code could be changed to just call floor once each for `q.x` and `q.y` and reuse the result here.

Reply:
if we remove the floor() we get pixel glitches at corner.
even we floor() we can also get glitches with specific float values.

I think we can forget about this patch